<launch>
    <!-- Arguments -->
    <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
    <arg name="map_file" default="$(find turtlebot3)/maps/map.yaml"/>
    <arg name="open_rviz" default="true"/>
    <arg name="move_forward_only" default="false"/>

    <!-- Turtlebot3 -->
    <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
        <arg name="model" value="$(arg model)" />
    </include>

    <!-- Run map server -->
    <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" output="screen"/>
    
    <!-- Run amcl node -->
    <include file="$(find turtlebot3_navigation)/launch/amcl.launch"/>

    <!-- move_base -->
    <include file="$(find turtlebot3)/launch/move_base.launch">
        <arg name="model" value="$(arg model)" />
        <arg name="move_forward_only" value="$(arg move_forward_only)"/>
    </include>

    <!-- rviz -->
    <group if="$(arg open_rviz)"> 
        <node pkg="rviz" type="rviz" name="rviz" required="true"
            args="-d $(find turtlebot3_navigation)/rviz/turtlebot3_navigation.rviz"/>
    </group>

    <!-- Start service server that responds with a plan for global path planning -->
    <node pkg="turtlebot3" name="bfs_server" type="bfs_server" output="screen"/> 
</launch>